doc/nrf/gs_recommended_versions.rst

```diff
@@ -0,0 +1,157 @@
+           -
+           - |ccache_recommended_ver_linux|
+         * - CMake
+           - |cmake_min_ver|
+           - |cmake_recommended_ver_linux|
+         * - dfu_util
+           -
+           - |dfu_util_recommended_ver_linux|
+         * - dtc
+           - |dtc_min_ver|
+           - |dtc_recommended_ver_linux|
+         * - git
+           -
+           - |git_recommended_ver_linux|
+         * - GNU Arm Embedded Toolchain
+           - |gnuarmemb_min_ver|
+           - |gnuarmemb_recommended_ver_linux|
+         * - gperf
+           - |gperf_min_ver|
+           - |gperf_recommended_ver_linux|
+         * - ninja
+           - |ninja_min_ver|
+           - |ninja_recommended_ver_linux|
+         * - Python
+           - |python_min_ver|
+           - |python_recommended_ver_linux|
+         * - SEGGER Embedded Studio (Nordic Edition)
+           - |ses_min_ver|
+           - |ses_recommended_ver_linux|
+
+   .. group-tab:: macOS
+
+      .. list-table::
+         :header-rows: 1
+
+         * - Tool
+           - Minimun version
+           - Toolchain manager version
+         * - CMake
+           - |cmake_min_ver|
+           - |cmake_recommended_ver_darwin|
+         * - dtc
+           - |dtc_min_ver|
+           - |dtc_recommended_ver_darwin|
+         * - git
+           -
+           - |git_recommended_ver_darwin|
+         * - GNU Arm Embedded Toolchain
+           - |gnuarmemb_min_ver|
+           - |gnuarmemb_recommended_ver_darwin|
+         * - gperf
+           - |gperf_min_ver|
+           - |gperf_recommended_ver_darwin|
+         * - ninja
+           - |ninja_min_ver|
+           - |ninja_recommended_ver_darwin|
+         * - Python
+           - |python_min_ver|
+           - |python_recommended_ver_darwin|
+         * - SEGGER Embedded Studio (Nordic Edition)
+           - |ses_min_ver|
+           - |ses_recommended_ver_darwin|
+
+
+Required Python dependencies
+############################
+
+The following table shows the python packages that are required for working with |NCS| |version|.
+If no version is specified, then the default version provided with pip is recommended.
+If a version is specified, then it is important that the version installed matches the required version.
+See :ref:`additional_deps` on how to install the Python dependencies.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Package
+     - Version
+   * - anytree
+     - |anytree_ver|
+   * - canopen
+     - |canopen_ver|
+   * - ecdsa
+     - |ecdsa_ver|
+   * - imagesize
+     - |imagesize_ver|
+   * - intelhex
+     - |intelhex_ver|
+   * - packaging
+     - |packaging_ver|
+   * - progress
+     - |progress_ver|
+   * - pyelftools
+     - |pyelftools_ver|
+   * - pylint
+     - |pylint_ver|
+   * - PyYAML
+     - |PyYAML_ver|
+   * - recommonmark
+     - |recommonmark_ver|
+   * - sphinxcontrib-mscgen
+     - |sphinxcontrib-mscgen_ver|
+   * - west
+     - |west_ver|
+   * - windows-curses (only Windows)
+     - |windows-curses_ver|
```

Comment:
I'm missing for example Sphinx and Breathe in this list (which are the ones that regularly give us trouble for the doc build).
Does it make sense to maintain this list "manually", only extracting the versions? Shouldn't the full list be generated?
When someone adds a new Python requirement, they will forget updating the table here ...

Reply:
> I'm missing for example Sphinx and Breathe in this list (which are the ones that regularly give us trouble for the doc build).

Because at this stage I only added the requirements for successfully building the projects.
If we want to also list the requirements for doc building, then I would suggest that goes into it's own table.

I think very few users of NCS wants to spend time building the docs themselves, when they can easily view them online.

> Does it make sense to maintain this list "manually", only extracting the versions? Shouldn't the full list be generated?

That would be nice, but then we would need to generate the rst file as well.
I would like to keep the table simple for this PR, but I agree, generating the table content would be nice in a future improvement.

> When someone adds a new Python requirement, they will forget updating the table here ...

Just like they might forget to add a package to `requirements.txt` in the first place:
https://github.com/nrfconnect/sdk-nrf/pull/2920

Reply:
There are already some of the doc tools in there (recommonmark, sphinxcontrib-mscgen), but not all. They can go into a different section if possible, sure, but since we document how to build the documentation, we also need to specify the requirements for it somewhere.

And imo you'll usually remember to add a requirement to the file that installs it ... but you will forget to document that it is needed.

Reply:
> There are already some of the doc tools in there (recommonmark, sphinxcontrib-mscgen), but not all.

Probably because nrf https://github.com/nrfconnect/sdk-nrf/blob/master/scripts/requirements.txt has not done the split between base and doc that Zephyr has:
- https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/requirements-base.txt
- https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/requirements-doc.txt

So either `recommonmark` and `sphinxcontrib-mscgen` should be moved to independent doc table, that also includes https://github.com/zephyrproject-rtos/zephyr/blob/master/scripts/requirements-doc.txt reqs, or they should be removed from this list, if they are solely needed for docs.

I've made an additional table, so people can comment.